### ISSUE TITLE
Plugin doesn't work with 2 windows opened in 3.4

### ISSUE THREAD

**Issue by wilsoniwano** [marcisme/sketch-preview#25]
Hi!

This great plugin doesn't work when I have two Sketch windows opened in 3.4 version.

Thanks!

**Comment by marcisme** [comment]
Thanks for reporting this issue.

Is it still a problem after closing and re-opening Sketch?

I have noticed that the plugin will sometimes stop working after changing the settings, but it works again after Sketch is re-launched. I'm still trying to figure out what the root problem is, but this may be the issue you are seeing. There will likely be a message in *Console.app* similar to the one below if this is the problem.

```
10/27/15 8:19:20.991 PM Preview (Sketch Plugin)[54716]: Caught script exception: *** -[__NSArrayM objectAtIndex:]: index 311 beyond bounds [0 .. 128]
```

**Comment by nitrada** [comment]
@marcisme for me, the plugin sometimes stops working after changing the settings. but … actually all plugins stop working :)

**Comment by nvasconcelos** [comment]
I'm also having the same problem on Sketch 3.4:
- Multiple windows
- Changing a setting from 3x to 2x

This is the log:
10/29/15 16:46:36.019 Preview (Sketch Plugin)[3003]: Caught script exception: *** -[__NSArrayM objectAtIndex:]: index 311 beyond bounds [0 .. 128]

**Comment by marcisme** [comment]
That's definitely the error I'm seeing. My best guess at this point is the Cocoascript pre-processor is somehow getting disabled, which would cause some of the shared code to no longer load. If that's the case, I may be able to inline the imported code as a temporary fix, but I'm still looking into it.

I'm really busy this week, but I'm going to make time to dig into this over the weekend. Thanks for your patience.

**Comment by marcisme** [comment]
I think this was a Sketch issue. There was a mention of a bug that affected plugins in the developer mailing list, and everything seems to work again in the latest beta.

If you can, give [Sketch 3.4.1 (15680)](https://rink.hockeyapp.net/apps/0172d48cceec171249a8d850fb16276b) a try, and let me know if there are still issues.

**Comment by leomondays** [comment]
Just to add some information to this thread: when I try to preview an artboard that height is superior to 1300, I got nothing but a black screen. I'm using Scala with 3x scale. Before this last Sketch update everything was working perfectly fine. :\

**Comment by leomondays** [comment]
Updating the information: I've tested this Sketch 3.4.1 version and the problem still occurs. When I have an artboard with a large height I got the black screen for the 3x scale in Preview Settings. For example: an artboard with height of 640, preview on Scala is fine. But with an artboard height of 1755 the preview goes all black. Any ideas?

**Comment by marcisme** [comment]
@leomondays I just tried this, and it's working for me. I used the *Desktop HD* artboard template and changed the dimensions to 1500 x 1755. I tried with the bare template with a colored background and also with a rectangle.

If you export the artboard manually, does that image look okay? Can you create a new, simpler Sketch document and try to reproduce with that? The Sketch document format has changed over time, so if it's an older document, that could be a factor too.

**Comment by marcisme** [comment]
These issues should be fixed in 3.4.2. Let me know if that's not the case.
